RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy

```diff
@@ -134,81 +134,69 @@ class BashConverter extends ConfigurationConverter {
         ]) {
             if (cfg.getConfigurationValues().getBoolean(bashFlag[0] as String, bashFlag[1] as Boolean)) text << separator << bashFlag[2] as String
         }
-        return text;
+        return text
     }
 
     StringBuilder appendPathVariables() {
-        StringBuilder text = new StringBuilder();
+        StringBuilder text = new StringBuilder()
 
         //Set a path if necessary.
         text << separator << new BashCommandSet().getSetPathCommand()
-        return text;
+        return text
     }
 
     /**
      *
-     * @param cfg    Configuration object. Basically a tree of configuration values that may additionally contain cross-references in the values.
+     * @param cfg Configuration object. Basically a tree of configuration values that may additionally contain cross-references in the values.
      * @return
      */
     Map<String, ConfigurationValue> getConfigurationValuesSortedByDependencies(Configuration cfg) {
```

Comment:
Still a pretty long method. Can it be broken down and made more descriptive? Why put everything into a single for-loop? Maybe use prefiltering to get rid of the useless stuff first?

```groovy
def boolean isValidationRule(String configurationName) {
   configurationName.contains("cfgValidationRule")
}

def boolean isComment(String configurationValue) {
   cv.value != null && cv.value.trim().startsWith("#")
}

values.findAll { !isValidationRule(it.id) }.findAll { it.value != null && !isComment(it.value) }
```

... then the rest.